### PR TITLE
fixing device only code that get called in the host side

### PR DESCRIPTION
**Pull request by mehdi-goli**

### CODE REVIEW

**Review by muhammad-tanvir-1211 — APPROVED** [review]

**Review by aacostadiaz — APPROVED** [review]
LGTM!

File: include/cutlass/cutlass.h [hunk 1]
```diff
@@ -115,205 +115,254 @@ static const int NumThreadsPerQuadPair = NumThreadsPerQuad * 2;
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 CUTLASS_HOST_DEVICE uint ThreadIdxX() {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__)
+  return threadIdx.x;
+#elif defined(__SYCL_Device_ONLY__)
```
Comment (AD2605) [inline include/cutlass/cutlass.h:120]: I believe the capitalization of `DEVICE` in `__SYCL_**Device**_ONLY__` is missing, should've been, `__SYCL_DEVICE_ONLY__`, 

I will open a PR ton fix this 👍🏻

File: include/cutlass/cutlass.h [hunk 2]
```diff
@@ -115,205 +115,254 @@ static const int NumThreadsPerQuadPair = NumThreadsPerQuad * 2;
...
 
 // math
 
 template <typename T>
 CUTLASS_DEVICE T hfma2(const T a, const T b, const T c) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__)
+  return hfma2(a, b, c);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
```
Comment (AD2605) [inline include/cutlass/cutlass.h:339]: @aacostadiaz , was there a reason `sycl::fma` was not used here ?
Acc. to the spec, It accepts sycl::vec of types float, double and half as well

PS - I am assuming that "Add SYCL equivalent function" means that there is no way to support this via sycl at the moment

Comment (aacostadiaz) [reply include/cutlass/cutlass.h:339]: We can use `sycl::fma` here. The TODO comment means let's figure out its equivalent later when we can properly test it.

**Pull request merged**